snake_test.go

```diff
@@ -135,7 +137,7 @@ func toDelimited(tb testing.TB) {
 		{"ManyManyWords", "many@many@words"},
 		{"manyManyWords", "many@many@words"},
 		{"AnyKind of_string", "any@kind@of@string"},
-		{"numbers2and55with000", "numbers@2@and@55@with@000"},
+		{"numbers2and55With000", "numbers2and55@with000"},
```

Comment:
And at this point, I'm almost sure it was intentional 😄

Reply:
It was, to show the change in behavior when a number is followed by a capital letter, but if it's confusing I'll change it back. At least in `toSnake` there are enough examples of that already. And keeping the tests the same better demonstrates the difference between this version and the last.